finality-aleph/src/justification/scheduler.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{MillisecsPerBlock, SessionPeriod};
+use std::{
+    cmp::min,
+    time::{Duration, Instant},
+};
+
+pub enum SchedulerActions {
+    ClearQueue,
+    Request,
+    Wait,
+}
+
+/// Bunch of methods for managing frequency of sending justification requests.
+pub trait JustificationRequestScheduler {
+    /// Decides whether we can request new justification.
+    fn schedule_action(&mut self) -> SchedulerActions;
+    /// Notice block finalization.
+    fn on_block_finalized(&mut self);
+    /// Notice request sending.
+    fn on_request_sent(&mut self);
+}
+
+pub struct JustificationRequestSchedulerImpl {
+    last_request_time: Instant,
+    last_finalization_time: Instant,
+    delay: Duration,
+    attempt: u32,
+    max_attemps: u32,
+}
+
+impl JustificationRequestSchedulerImpl {
+    pub fn new(
+        session_period: &SessionPeriod,
+        millisecs_per_block: &MillisecsPerBlock,
+        max_attemps: u32,
+    ) -> Self {
+        Self {
+            last_request_time: Instant::now(),
+            last_finalization_time: Instant::now(),
+            delay: Duration::from_millis(min(
```

Comment:
a comment for why these constants would be nice here

Reply:
Those are values that @DamianStraszak come out with a few months ago, I didn't change anything here. In other words I don't really know why these constants were chosen xd

Reply:
There is not much I can say :) You must have some delays here so that
1) you don't check too often
2) you check often enough so that the latency for non-validators is small.

Checking every `2` blocks sounds right to me. 

The min is in case someone does some peculiar tests with session_period like `2`. If the other branch was not there (in the `min`) then the non-validator nodes would lag behind.